### PR TITLE
Tree view

### PR DESCRIPTION
**Pull request by asvae**
Notables:
* Add docs as .md file. Easier to edit that way. These are duplicated in wiki page.
* Add yarn.lock file.
* Library probably requires a bit or external testing. DX wise.
* Fix icons demo #146
* (Temporary) disable eslint comma-dangle rule. Was Infuriating.

### CODE REVIEW

File: src/components/ui/icons/Set.vue [hunk 1]
```diff
@@ -58,11 +58,10 @@
...
                   :style="`font-size: ${iconSize}px`"
                   aria-hidden="true"
-                >
-                </span>
-                <div class="iconText">
-                  {{ list.icons[(i - 1) * 8 + j - 1] }}
-                </div>
+                />
+              </div>
+              <div class="iconText">
```
[the remark this answers is not in the document]
Comment (asvae) [reply src/components/ui/icons/Set.vue]: Heizenbug.

File: src/vuestic-theme/vuestic-components/vuestic-tree-view/vuestic-div-checkbox/VuesticDivCheckbox.vue [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+  }
+</script>
+
+<style lang="scss">
+  @import "../../../../sass/variables";
+
+  .vuestic-div-checkbox {
+    cursor: pointer;
+    &__icon-hollow {
```
Comment (smartapant) [inline src/vuestic-theme/vuestic-components/vuestic-tree-view/vuestic-div-checkbox/VuesticDivCheckbox.vue]: no bem please, rename everywhere

File: src/vuestic-theme/vuestic-components/vuestic-tree-view/vuestic-div-checkbox/VuesticDivCheckbox.demo.vue [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<template>
```
Comment (smartapant) [inline src/vuestic-theme/vuestic-components/vuestic-tree-view/vuestic-div-checkbox/VuesticDivCheckbox.demo.vue]: please, fix vuestic-checkbox instead

File: src/components/ui/tree-view/TreeViewRootFactory.js [hunk 4]
```diff
@@ -0,0 +1,200 @@
+// export default class TreeViewRootFactory {
```
Comment (smartapant) [inline src/components/ui/tree-view/TreeViewRootFactory.js]: delete this file :)

File: src/components/ui/tree-view/TreeView.vue [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+          </vuestic-widget>
+        </div>
+
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+  import VuesticWidget
```
Comment (smartapant) [inline src/components/ui/tree-view/TreeView.vue]: delete these imports

File: src/components/ui/tree-view/TreeView.vue [hunk 6]
```diff
@@ -0,0 +1,100 @@
+<template>
```
Comment (smartapant) [inline src/components/ui/tree-view/TreeView.vue]: play with layout here (maybe swapping some of the trees could help)

File: src/components/ui/tree-view/TreeView.vue [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+              </div>
+            </div>
+          </vuestic-widget>
+        </div>
+
+        <div class="small-set col-lg-6">
+          <vuestic-widget :headerText="$t('treeView.selectable')">
+            <div class="set-content">
+              <div class="overlay">
+                <tree-view-selectable-preview/>
```
Comment (smartapant) [inline src/components/ui/tree-view/TreeView.vue]: selection class missing

File: src/components/ui/tree-view/TreeViewEditablePreview.vue [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+            <i class="bar"></i>
+          </div>
+        </div>
+        <div slot="iconRight" class="icon" @click="removeProduct(product)">
+          <span aria-hidden="true"
+                class="ion ion-md-close"/>
+        </div>
+      </vuestic-tree-node>
+      <vuestic-tree-node>
+        <button class="btn btn-primary btn-micro"
```
Comment (smartapant) [inline src/components/ui/tree-view/TreeViewEditablePreview.vue]: ban's box-shadow is cropped

File: src/components/ui/tree-view/TreeViewEditablePreview.vue [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    <vuestic-tree-category isOpen label="Products">
+      <vuestic-tree-node v-for="product in products" :key="product.id">
+        <div class="form-group">
+          <div class="input-group">
+            <input v-model="product.name">
+            <i class="bar"></i>
+          </div>
+        </div>
+        <div slot="iconRight" class="icon" @click="removeProduct(product)">
+          <span aria-hidden="true"
```
Comment (smartapant) [inline src/components/ui/tree-view/TreeViewEditablePreview.vue]: spans indentations are crazy

File: src/vuestic-theme/vuestic-components/vuestic-tree-view/vuestic-tree-view-readme.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
+```html
```
Comment (smartapant) [inline src/vuestic-theme/vuestic-components/vuestic-tree-view/vuestic-tree-view-readme.md]: rm the file and add this to wiki

File: .eslintrc.js [hunk 11]
```diff
@@ -17,6 +17,7 @@ module.exports = {
   ],
   // add your custom rules here
   'rules': {
+    'comma-dangle': 'off',
```
Comment (smartapant) [inline .eslintrc.js]: revert this rule (on)

File: build/utils.js [hunk 12]
```diff
@@ -25,6 +25,7 @@ exports.cssLoaders = function (options) {
   function generateLoaders (loader, loaderOptions) {
     const loaders = [cssLoader]
     if (loader) {
+
```
Comment (smartapant) [inline build/utils.js]: why blank line here?

File: src/components/ui/tree-view/TreeView.vue [hunk 13]
```diff
@@ -0,0 +1,100 @@
+<template>
+  <div class="sets-list row">
+    <div class="col-md-12">
+      <div class="row">
+
+        <div class="small-set col-lg-6">
+          <vuestic-widget :headerText="$t('treeView.basic')">
+            <div class="set-content">
```
Comment (smartapant) [inline src/components/ui/tree-view/TreeView.vue]: remove useless wrappers (set-conent, overlay)

File: src/vuestic-theme/vuestic-components/vuestic-tree-view/vuestic-div-checkbox/VuesticDivCheckbox.vue (hunk 2 above)
Comment (asvae) [reply src/vuestic-theme/vuestic-components/vuestic-tree-view/vuestic-div-checkbox/VuesticDivCheckbox.vue]: Done

File: src/components/ui/tree-view/TreeViewRootFactory.js (hunk 4 above)
Comment (asvae) [reply src/components/ui/tree-view/TreeViewRootFactory.js]: Done

File: src/components/ui/tree-view/TreeView.vue (hunk 5 above)
Comment (asvae) [reply src/components/ui/tree-view/TreeView.vue]: Done

File: src/components/ui/tree-view/TreeViewEditablePreview.vue (hunk 9 above)
Comment (asvae) [reply src/components/ui/tree-view/TreeViewEditablePreview.vue]: Done

File: src/vuestic-theme/vuestic-components/vuestic-tree-view/vuestic-tree-view-readme.md (hunk 10 above)
Comment (asvae) [reply src/vuestic-theme/vuestic-components/vuestic-tree-view/vuestic-tree-view-readme.md]: Done

File: src/components/ui/tree-view/TreeView.vue (hunk 13 above)
Comment (asvae) [reply src/components/ui/tree-view/TreeView.vue]: Done

File: build/utils.js (hunk 12 above)
Comment (asvae) [reply build/utils.js]: Done

File: .eslintrc.js (hunk 11 above)
Comment (asvae) [reply .eslintrc.js]: Disagreement on this. Discuss.

File: src/components/ui/tree-view/TreeViewEditablePreview.vue (hunk 8 above)
Comment (asvae) [reply src/components/ui/tree-view/TreeViewEditablePreview.vue]: Done

File: .eslintrc.js (hunk 11 above)
Comment (asvae) [reply .eslintrc.js]: Agreement reached on disabling ATM.

File: src/vuestic-theme/vuestic-components/vuestic-tree-view/vuestic-div-checkbox/VuesticDivCheckbox.demo.vue (hunk 3 above)
Comment (asvae) [reply src/vuestic-theme/vuestic-components/vuestic-tree-view/vuestic-div-checkbox/VuesticDivCheckbox.demo.vue]: Done

**Pull request merged**